pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -410,10 +413,10 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 	// Sets the v1.NodeNetworkUnavailable condition to False.
 	ca.setNetworkCondition(node)
 
-	// Nodes in clusters WITHOUT multi-networking are expected to have only 1 network-interface
-	// with 1 alias IPv4 range and/or 1 IPv6 address. Multi-network clusters may have 1 interface
-	// with multiple aliases.
-	if len(instance.NetworkInterfaces) == 1 &&
+	// When multi-networking is disabled, the IPAM logic only considers the primary
+	// network interface and ignores any additional network interfaces. 
+	// The primary network interface is expected to have at most one IPv4 alias and/or one IPv6 address.
+	if !ca.enableMultiNetworking &&
 		(len(instance.NetworkInterfaces[0].AliasIpRanges) == 1 ||
 			ca.cloud.GetIPV6Address(instance.NetworkInterfaces[0]) != nil) {
```

Comment:
For future proofing, a case here that we may now need to configure is whether the interface at index 0 is the default one or not. Our logic in this conditional assumes that it's working with the default interface and we'd like it to to continue doing so. 

As per the compute engine API, https://cloud.google.com/compute/docs/reference/rest/v1/instances/get 

_"The default interface value is nic0."_

<img width="901" height="111" alt="image" src="https://github.com/user-attachments/assets/999ceb11-4982-49fa-9c1c-08540b4ec37a" />

This could potentially be used to figure out the interface which we have to work with.